### PR TITLE
Add 'not widely supported' SGR to support rate.cx

### PR DESCRIPTION
**Pull request by yne**

### CODE REVIEW

**Comment by yne** [comment]
It should look like this

**Comment by yne** [comment]
Also there is a unused `prx` branch that shall be removed

File: net_http_bsd/src/main.c [hunk 1]
```diff
@@ -49,38 +55,67 @@ unsigned char readUTF(unsigned char c, int fd){
...
+		if (c == '\n') psvDebugScreenPrintf("%.*s\n",pos,line);
 		else line[pos] = c;
 	}
-
 	close(fd);
+
+	#ifndef __vita__ // generate a RGB screen dump if built on PC
+	//convert the dump into PNG with: convert -depth 8 -size 960x544+0 RGB:screen.data out.png;
+	int fdump = open("screen.data", O_WRONLY | O_CREAT | O_TRUNC, 0777);
+	for (int i=0;i<(int)sizeof(base);i += 4)write(fdump, base+i, 3);
```
Comment (devnoname120) [inline net_http_bsd/src/main.c]: Please expand this loop. It's hard to read, especially for beginner developers who are probably the main target of the samples.

File: net_http_bsd/src/main.c [hunk 2]
```diff
@@ -49,38 +55,67 @@ unsigned char readUTF(unsigned char c, int fd){
...
+	case 9607:return 0xDB;
+	case 9608:return 0xDB;
 	}
 	return '?';
 }
 
 int main (int argc, char *argv[]){
-	static char net_mem[1*1024*1024];
+	(void)argc;(void)argv;
+	char* *url=NULL, *wttr[]={"wttr.in","/Paris"}, *rate[]={"rate.sx","/"};
```
Comment (devnoname120) [inline net_http_bsd/src/main.c]: Please don't compact your variables declarations so much, it makes reading them harder and it doesn't play well with diff tools.

File: net_http_bsd/src/main.c [hunk 3]
```diff
@@ -49,38 +55,67 @@ unsigned char readUTF(unsigned char c, int fd){
...
-	connect(fd, (const struct sockaddr *)&((struct sockaddr_in){.sin_family = AF_INET, .sin_port = htons(80), .sin_addr.s_addr = *(long*)(gethostbyname(host)->h_addr)}), sizeof(struct sockaddr_in));
+	connect(fd, (const struct sockaddr *)&((struct sockaddr_in){.sin_family = AF_INET, .sin_port = htons(80), .sin_addr.s_addr = *(long*)(gethostbyname(url[0])->h_addr)}), sizeof(struct sockaddr_in));
 	
-	char*header[] = {"GET ",path," HTTP/1.1\r\n", "User-Agent: curl/7.52.1\r\n", "Host: ",host,"\r\n", "\r\n", 0};
+	char*header[] = {"GET ",url[1]," HTTP/1.1\r\n", "User-Agent: curl/PSVita\r\n", "Host: ",url[0],"\r\n", "\r\n", 0};
 	for(int i=0;header[i];i++)write(fd, header[i], strlen(header[i]));
 
-	for(unsigned char c, line[4096];read(fd,&c,sizeof(c))>0 && pos<sizeof(line);pos = (c=='\n')? 0 : (pos + 1)){
-		#ifdef __vita__
+	for(unsigned char c, line[4096];read(fd,&c,sizeof(c))>0 && pos<(int)sizeof(line);pos = (c=='\n')? 0 : (pos + 1)){
```
Comment (devnoname120) [inline net_http_bsd/src/main.c]: Please expand this as well.

**Comment by yne** [comment]
I think It's okay now :)

**Pull request merged**